### PR TITLE
Created Upgrades Tree

### PR DESCRIPTION
**Pull request by Ido-Barnea**

### CODE REVIEW

File: core/development/Game.ts [hunk 1]
```diff
@@ -104,12 +104,13 @@ function endMove(canRecover = true) {
 function checkForUpgradeablePawns() {
   for (let index = 0; index < pieces.length; index++) {
     const piece = pieces[index];
-    if (piece instanceof Pawn) {
+    if (piece instanceof Pawn && piece.position) {
       const whitePawnReachedEndOfBoard =
-        piece.player === whitePlayer && piece.position.coordinates[1] === 0;
+        piece.player === whitePlayer && piece.position?.coordinates[1] === 0;
```
Comment (Monkey-bored) [inline core/development/Game.ts:109]: Do we need the ? after piece.position if the if already checks that piece.position is not undefined?

File: core/development/ui/UpgradeUI.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  piece: Piece,
+  upgradeablePieces: Array<new (player: Player) => Piece>,
+) {
+  const upgradesContainer = document.getElementById('piece-upgrades-container');
+  if (!upgradesContainer) return;
+
+  upgradesContainer.innerHTML = '';
+
+  upgradeablePieces.forEach((upgradeablePieceType) => {
+    const upgradeablePiece = new upgradeablePieceType(game.getCurrentPlayer());
```
Comment (Monkey-bored) [inline core/development/ui/UpgradeUI.ts:16]: I think this should preferably not be in the UI section

File: core/development/Game.ts (hunk 1 above)
Comment (Ido-Barnea) [reply core/development/Game.ts:109]: We do not

File: core/development/ui/UpgradeUI.ts (hunk 2 above)
Comment (Ido-Barnea) [reply core/development/ui/UpgradeUI.ts:16]: If we want to access `name` and `resource`, it needs to be

**Review by Monkey-bored — APPROVED** [review]
### LGTM 😁😁😁

**Pull request merged**